ztrouy/game-rater-api
# Feature - Games List View

### Purpose
To allow Users to view a list of all Games

### Files Changed
- Created file, and wrote serializer for Categories in new `categories.py`
- Created file, wrote serializer for Games, and created GET method for returning list of Games in new `games.py` View Set
- Imported Game and User View Sets in `__init__.py` package module in the `/views` directory
- Added routing for User and Game View Sets in `urls.py`
- Added valid property to response in the login method in `users.py`
- Setup base client application in `/client` directory
- Created file, and wrote rendering and logic for navbar in new `NavBar.jsx`
- Created file, and wrote fetches for logging in and registering in new `userManager.js`
- Created file, and wrote rendering and logic for Login view in new `Login.jsx`
- Created file, and wrote rendering and logic for Registration view in new `Register.jsx`
- Created file, and wrote rendering for a Game card in new `GameCard.jsx`
- Created file, and wrote rendering and logic for displaying a list of all Games in new `GameList.jsx`
- Created file, and wrote rendering for Games List view in new `Games.jsx`
- Added placeholder routing application, and functional routing for Login, Register, and Games List views in `ApplicationViews.jsx`

### To Test
Fetch, setup, create database using `seed_database.sh`, run debugger for backend, and run client using `npm run dev` in client directory, then confirm the following
- GET endpoint `/games` returns a list of all Games when authenticated
- Users can only visit Login and Regster views when not logged in
- Logged in Users are directed to placeholder Home view
- Clicking the "Logout" button in the navbar logs the User out
- Clicking the "Games" button in the navbar navigates the User to the Games List view
- Games in the Games List view display their Title, Designer, Description, and Categories

closes #2
